### PR TITLE
[PATCH] [EDA-1581] Adding scenarios use case

### PR DESCRIPTION
**Pull request by mbertani-eb**
Add missing scenarios of use cases that are essential  to ensure the performance of the game
I used  two functions  `element_for_shoot_action_test` `generate_board_for_move_action_test` to setup the scenarios for the tests.
( I refactor by the way the `element_for_shoot_action_test` function to be more readable)

I created a new test `test_execute_action_invalid` for the invalid moves.

I added to the `Cell` 4 methods, 
 - 2 of them are for overwriting the property inherited method of `TreasureHolder` for `golds` and `diamonds`, to take count of the treasures of the character over the cell
 - the other two are to indicate if in the cell there is gold or diamond, either because of the cell or the character over the cell

### CODE REVIEW

**Review by npaura-eb — APPROVED** [review]

File: constans/scenarios.py [hunk 1]
```diff
@@ -556,31 +556,103 @@ def board_player_1_scenario():
...
-CELL_SHOOT_8_8.is_discover[0] = True
+def generate_board_for_shoot_action_test():
+    scenario = [[Cell(i, j) for j in range(LARGE)] for i in range(LARGE)]
+
+    shooter_player = Player(PLAYER_1)
+    shooted_player = Player(PLAYER_2)
+
+    # add one more character to test cases
+    shooter_player.characters.append(Character(shooter_player))
+    shooted_player.characters.append(Character(shooted_player))
```
Comment (ceusebi-eb) [inline constans/scenarios.py:569]: Past tense of `shoot` is `shot`. You will have to change this across the whole PR

File: game/cell.py [hunk 2]
```diff
@@ -63,12 +63,30 @@ def _middle_char(self, representation: list):
...
+                else self.gold)
+
+    def there_is_diamond(self):
+        return (self.diamond or self.character.diamond
+                if self.character
+                else self.diamond)
+
+    @TreasureHolder.gold.getter
+    def gold(self):
+        return super().gold + self.character.gold if self.character else super().gold
```
Comment (ceusebi-eb) [inline game/cell.py:88]: I feel this expression conveys the meaning better
```suggestion
        return super().gold + (self.character.gold if self.character else 0)
```

File: game/cell.py [hunk 3]
```diff
@@ -63,12 +63,30 @@ def _middle_char(self, representation: list):
...
+
+    @TreasureHolder.diamond.getter
+    def diamond(self):
+        return super().diamond + self.character.diamond if self.character else super().diamond
```
Comment (ceusebi-eb) [inline game/cell.py:92]: Same as above

File: game/cell.py (hunk 2 above)
Comment (mbertani-eb) [reply game/cell.py]: Yes, It's more readable and elegant. Thank you

**Pull request merged**